### ISSUE TITLE
Find commits with one or multiple parents that altered a specific file

### ISSUE THREAD

**Issue by 93Sprinter** [Murmele/Gittyup#23]
Hi,
I'm not sure if I use the history search of Gittyup correctly but if I try to find all commits, that altered a specific file, the search result only contains commits with one parent. e.g. commits with multiple parents like merge commits are not listed.

Is it possible to find those commits just like if I use "git log --simplify-merges" or "git log --first-parent"?

**Comment by Murmele** [comment]
It seems that also path does not work correctly or is it for you?

**Comment by 93Sprinter** [comment]
Yes, I also have the same problem with the path option. Only commits with parent are shown.

**Comment by Murmele** [comment]
hi @93Sprinter I will have a look on it. Thanks for the hint

**Comment by Murmele** [comment]
Where did you see that? Can you say me the file and line?

**Comment by 93Sprinter** [comment]
The switch statement is in file "src/git/RevWalk.cpp". The method starts at line 58, the switch statement starts at line 82.
And I think that this method is called in "src/index/indexer.cpp". At line 388 the walker is created and at line 422 begins a while loop where the commits for indexing are collected. There is also a comment which stats that merge commits are not indexed.

**Comment by Murmele** [comment]
Thank you for indicating me to the correct place. Not sure yet when I come to it

**Comment by AHSauge** [comment]
This is going to be quite a lot easier to fix when libgit2 v2.0 is released, since it would include https://github.com/libgit2/libgit2/pull/7281, which adds an API that would obsolete a lot of the code in `RevWalk::next`.
